subsys/fs/littlefs_fs.c

```diff
@@ -842,8 +842,25 @@ static int littlefs_init_cfg(struct fs_littlefs *fs, int flags)
 
 		lcp->read_size = block_size;
 		lcp->prog_size = block_size;
-		lcp->cache_size = block_size;
-		lcp->lookahead_size = block_size * 4;
+
```

Comment:
Issues:
 - in line 837 there is statement that here we are setting "validated/defaulted" values, which means that this should not be happening here
 - this is hidden logic that fixes problem by leading us into next where mounting of LFS takes away heap, and make LFS or other subsystems fail, depending on the order of execution.

I think the proper solution here would be to fail initialization each time when condition happens, with LOG, because in that case we should have stable path of execution that uniformly warns user on misconfiguration.

Reply:
Failing with a LOG indicating misconfiguration makes sense to me, thanks!  I'll make that change and update the PR after I try it out on my setup.

Reply:
I changed it to use LOG_ERR and return when this situation is encountered.  Example output with logging enabled:

```
$ fs mount littlefs /stuff
Error mounting as littlefs: -12
[00:00:15.760,000] <inf> littlefs: LittleFS version 2.9, disk version 2.1
[00:00:15.940,000] <inf> littlefs: FS at SDMMC: is 524288 0x200-byte blocks with 512 cycle
[00:00:15.940,000] <err> littlefs: Configured cache size is too small: 64 < 512
[00:00:15.940,000] <err> littlefs: Configured lookahead size is too small: 32 < 2048
[00:00:15.940,000] <err> fs: fs mount error (-12)
```